pkg/reconciler/jbsconfig/jbsconfig.go

```diff
@@ -60,6 +60,42 @@ func (r *ReconcilerJBSConfig) Reconcile(ctx context.Context, request reconcile.R
 	jbsConfig := v1alpha1.JBSConfig{}
 	err := r.client.Get(ctx, request.NamespacedName, &jbsConfig)
 	if err != nil {
+		// Deleted JBSConfig - delete cache resources
+		if errors.IsNotFound(err) && request.Name == v1alpha1.JBSConfigName {
+			service := &corev1.Service{}
+			err = r.client.Get(ctx, types.NamespacedName{Name: v1alpha1.CacheDeploymentName, Namespace: request.Namespace}, service)
```

Comment:
I would just delete straight away, and then check if you got a not found error, where you just move on if you did get a not found error, and but then log otherwise

I'd also create an warning event if we could not delete any of the cache related artifacts 

do this for all the items below

Reply:
added the warning events on objects which having issue with deletion.

Reply:
yes sorry that was unclear .... I mean on the object being deleted

I think the other part of my comment also was not clear (just delete vs. get/delete).  Let my try a code snippet to clear up:

```
service := &corev1.Service{}
service.Name = v1alpha1.CacheDeploymentName
service.Namespace = request.Namespace
err = r.client.Delete(ctx, service)
if err != nil && !errors.IsNotFound(err) {
   // log error and record event
}
// otherwise just continue
```

IIRC there are no resource version checks or anything like that on the delete.  I just works off of the name/namespace.

Reply:
thanks, updated.